spec/core/output_sink/base_sink_spec.rb

```diff
@@ -2,12 +2,162 @@
 
 describe Core::OutputSink::BaseSink do
   subject { described_class.new }
+  let(:document) { { :id => 15 } }
+  let(:deleted_id) { 25 }
 
   it 'base_sink methods raise an error when called' do
-    expect { subject.ingest(nil) }.to raise_error('not implemented')
-    expect { subject.ingest_multiple(nil) }.to raise_error('not implemented')
-    expect { subject.delete(nil) }.to raise_error('not implemented')
-    expect { subject.delete_multiple(nil) }.to raise_error('not implemented')
-    expect { subject.flush(_size: nil) }.to raise_error('not implemented')
+    expect { subject.ingest(document) }.to raise_error(NotImplementedError)
+    expect { subject.ingest_multiple([document]) }.to raise_error(NotImplementedError)
+    expect { subject.delete(deleted_id) }.to raise_error(NotImplementedError)
+    expect { subject.delete_multiple([deleted_id]) }.to raise_error(NotImplementedError)
+    expect { subject.flush }.to raise_error(NotImplementedError)
+  end
+
+  context 'when all private methods are implemented' do
+    before(:each) do
+      allow(subject).to receive(:do_ingest)
+      allow(subject).to receive(:do_delete)
+      allow(subject).to receive(:do_flush)
+      allow(subject).to receive(:do_serialize)
+    end
+
+    context '#ingest' do
```

Comment:
[nitpick] It's better to use `#describe` when you start testing a method. 

refs: https://www.betterspecs.org

Reply:
Good call, but I'll make it in a different PR for all tests in ruby project - they all don't follow this. Also some `#` or `.` in method prefixes don't make much sense, I'll fix that in a separate PR as well.